cognite/extractorutils/uploader.py

```diff
@@ -93,10 +94,10 @@
 RETRY_DELAY = 5
 RETRIES = 10
 
-DataPointList = Union[
-    List[Dict[Union[int, float, datetime], Union[int, float, str]]],
-    List[Tuple[Union[int, float, datetime], Union[int, float, str]]],
+DataPoint = Union[
+    Dict[str, Union[int, float, str, datetime]], Tuple[Union[int, float, datetime], Union[int, float, str]]
 ]
+DataPointList = List[DataPoint]
```

Comment:
I believe this is too generic, it would allow e.g.

```
[(ts1, val1), {ts2: val2}]
```

which I don't think the SDK allows?

Reply:
This is more generic though, so were not breaking anything, but I believe there are things that would type check and then break when run - but I actually think that might already be the case for the previous type hint.

Maybe we should just make it much simpler. Just doing a list of tuples or something like that, and de-generify it a little bit. That might be nice for readability and keeping it understandable. That's a breaking change though, so maybe merge this as-is, and then do a redesign of this whole typing situation for a 2.0.0 release.

Reply:
Urgh. You're right. It's a `List[Union]` vs `Union[List]` thing again... I guess we'll merge it as is.